code_review_agent/main.py

```
@@ -0,0 +1,247 @@
+        
+    except Exception as e:
+        logger.error(
+            "Error handling GitHub webhook",
+            error=str(e),
+            exc_info=True
+        )
+        _metrics["total_errors"] += 1
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Webhook processing failed: {str(e)}"
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)